app/controllers/users/reset_passwords_controller.rb

```diff
@@ -83,6 +100,11 @@ def handle_unsuccessful_password_reset(result)
       render :edit
     end
 
+    def update_user
+      resource.update(confirmed_at: Time.current) unless resource.confirmed?
```

Comment:
I'm not sure that we can completely remove Devise, but I agree that for controllers that don't really need to inherit from Devise, we can remove reference to `resource`. I'd be happy to clean up this controller in a follow up PR.